zebedee-cms/src/test/java/com/github/onsdigital/zebedee/api/ContentTest.java

```diff
@@ -0,0 +1,275 @@
+
+        when(mockSessions.get()).thenReturn(mockSession);
+
+        when(mockCollectionKeyring.get(any(), any())).thenReturn(secretKey);
+
+        when(mockZebedee.getCollections()).thenReturn(collections);
+        when(mockZebedee.getSessions()).thenReturn(mockSessions);
+        when(mockZebedee.getEncryptionKeyFactory()).thenReturn(encryptionKeyFactory);
+        when(mockZebedee.getPermissionsService()).thenReturn(mockPermissionsService);
+        when(mockZebedee.getUsersService()).thenReturn(mockUsersService);
+        when(mockZebedee.getCollectionKeyring()).thenReturn(mockCollectionKeyring);
+        when(mockZebedee.getSlackNotifier()).thenReturn(mockNotifier);
+        Root.zebedee = mockZebedee;
+    }
+
+    @Test
+    public void WriteVersionFileV1() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        String collectionId = "aktesting";
+        request.setPathInfo("/content/" + collectionId);
+        String datasetURL = "/peoplepopulationandcommunity/birthsdeathsandmarriages/livebirths/datasets/babynamesenglandandwalesbabynamesstatisticsboys/2022/data.json";
+        request.addParameter("uri", datasetURL);
+        request.setContent(getV1RequestContent());
+
+        Path versionFile = createCollectionAndPaths(collectionId);
+
+        // Create the version content via the content API.
+        try {
+            Content contentApi = new Content();
+            boolean result = contentApi.saveContent(request, new MockHttpServletResponse());
+            Assert.assertTrue(result);
+        } catch (Exception e) {
+            Assert.fail(e.getMessage() + e.toString());
+            e.printStackTrace();
+            throw e;
+        }
+
+        byte[] bytes = IOUtils.toByteArray(EncryptionUtils.encryptionInputStream(versionFile, secretKey));
+        Assert.assertEquals(expectedContentV1, new String(bytes));
+
+        // read the version content via the content API.
+       MockHttpServletResponse response = new MockHttpServletResponse();
+        MockHttpServletRequest dataRequest = new MockHttpServletRequest();
+        dataRequest.setRequestURI("/data/" + collectionId);
+
+        String dataURL = "/peoplepopulationandcommunity/birthsdeathsandmarriages/livebirths/datasets/babynamesenglandandwalesbabynamesstatisticsboys/2022";
+        dataRequest.addParameter("uri", dataURL);
+
+        ZebedeeCollectionReaderFactory factory = new ZebedeeCollectionReaderFactory(mockZebedee);
+        ZebedeeReader.setCollectionReaderFactory(factory);
+
+        try {
+            Data dataAPI = new Data();
+            dataAPI.read(dataRequest, response);
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.fail(e.getMessage() + e.toString());
+            throw e;
+        }
+
+        Assert.assertEquals(200, response.getStatus());
+        JSONAssert.assertEquals(expectedContentV1, response.getContentAsString(), false);
```

Comment:
I would try and avoid adding a new dependency for this. Maybe we could create our own assertion as it should be as simple as

```
    private void assertJsonEquals(String expected, String actual) {
        Assert.assertEquals(Json.parseJson(expected), Json.parseJson(actual));
    }

```